release_notes/ocp-4-5-release-notes.adoc

```diff
@@ -223,10 +223,23 @@ xref:../networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.ad
 
 There are two noteworthy Ingress enhancements introduced in {product-title} 4.5:
 
-* You can enable access logs for the Ingress Controller.
-//:../networking/ingress-operator.adoc#nw-configure-ingress-access-logging_configuring-ingress[enable access logs for the Ingress Controller].
+* You can xref:../networking/ingress-operator.adoc#nw-configure-ingress-access-logging_configuring-ingress[enable access logs for the Ingress Controller].
 * You can xref:../networking/ingress-operator.adoc#using-wildcard-routes_configuring-ingress[specify a wildcard route policy through the Ingress Controller].
 
+[id="ocp-4-5-haproxy-upgraded"]
+==== HAProxy upgraded to version 2.0.14
+
+The HAProxy used for Ingress has been upgraded from version 2.0.13 to 2.0.14.
+This upgrade provides a router reload performance improvement. The router reload
+optimization is most beneficial for clusters with thousands of routes.
+
+[id="ocp-4-5-http-2-ingress-support"]
+==== HTTP/2 Ingress support
+
+{product-title} 4.5 enables transparent end-to-end HTTP/2 support in HAProxy for
+application owners to make use of its protocol capabilities, including single
+connection, header compression, binary streams, and more.
```

Comment:
We should mention the following caveats:

* To enable HTTP/2 for the connection from the client to HAProxy, a route must specify a custom certificate.  A route that uses the default certificate cannot use HTTP/2.  This restriction is necessary to avoid problems from connection coalescing (where the client re-uses a connection for different routes that use the same certificate).

* The connection from HAProxy to the application pod can use HTTP/2 only for reencrypt routes and not for edge-terminated or insecure routes.  This restriction comes from the fact that HAProxy uses ALPN (Application-Level Protocol Negotiation), which is a TLS extension, to negotiate the use of HTTP/2 with the backend.  This restriction also applied in 4.4.  For 4.5, the implication is that end-to-end HTTP/2 is possible with passthrough and reencrypt and not with insecure or edge-terminated routes.

@frobware, is there anything else to add?

Reply:
@Miciah @frobware I have added the above recommendations. Please confirm the current revision of the release note is adequate. Thanks!

Reply:
> We should mention the following caveats:
> 
> * To enable HTTP/2 for the connection from the client to HAProxy, a route must specify a custom certificate.  A route that uses the default certificate cannot use HTTP/2.  This restriction is necessary to avoid problems from connection coalescing (where the client re-uses a connection for different routes that use the same certificate).
> * The connection from HAProxy to the application pod can use HTTP/2 only for reencrypt routes and not for edge-terminated or insecure routes.  This restriction comes from the fact that HAProxy uses ALPN (Application-Level Protocol Negotiation), which is a TLS extension, to negotiate the use of HTTP/2 with the backend.  This restriction also applied in 4.4.  For 4.5, the implication is that end-to-end HTTP/2 is possible with passthrough and reencrypt and not with insecure or edge-terminated routes.
> 
> @frobware, is there anything else to add?

No, this is good. Thanks!